### PR TITLE
[Graph] Fix CompiledGraph aliasing bug

### PR DESCRIPTION
**Pull request by Aalanli**
Some graphs produce output tensors that alias. In the original case, the outputs of the aliasing tensors would be undefined (the result of empty). This PR fixes that by aliasing the outputs as well.

For example this script originally outputs garbage in the second tensor
```
import hidet

a = hidet.symbol([1, 2], device='cuda')
b = a.mean(1)
g = hidet.trace_from([b, b], a)

cg = g.build()
print(cg(hidet.ones([1, 2], device='cuda'))[1])
```

### CODE REVIEW

File: python/hidet/runtime/compiled_graph.py [hunk 1]
```diff
@@ -208,6 +208,10 @@ def _create_outputs(self, inputs):
 
         outputs = []
         for output_index, (exec_idx, sig) in enumerate(zip(self.graph_execution.outputs_index, self.meta.outputs)):
+            if sig.alias is not None:
+                outputs.append(outputs[sig.alias])
+                continue
+
```
Comment (yaoyaoding) [inline python/hidet/runtime/compiled_graph.py:214]: We can do the checking alternatively like
```
            if exec_idx in self.graph_execution.inputs_index:
                outputs.append(inputs[self.graph_execution.inputs_index.index(exec_idx)])
            elif exec_idx in self.graph_execution.weights_index:
                outputs.append(self.weights[self.graph_execution.weights_index.index(exec_idx)])
            elif exec_idx != self.graph_execution.outputs_index.index(exec_idx): # <--------
                outputs.append(outputs[self.graph_execution.outputs_index.index(exec_idx)])
            else:
```
so we do not need to modify the graph building process or an alias attribute.

How do you think @Aalanli ?

**Review by yaoyaoding — COMMENTED** [review]
Hi @Aalanli,

Thanks for the fix! There is an alternative way to fix this that you can consider. Besides, let's add the tests provided by the @Azyka to our test bed.

File: commutative_law.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
+# %%
```
Comment (yaoyaoding) [inline commutative_law.py:1]: Remove this file.

File: python/hidet/runtime/compiled_graph.py (hunk 1 above)
Comment (Aalanli) [reply python/hidet/runtime/compiled_graph.py]: The suggested change doesn't work, but I see how to fix it. 
Yeah, I think this way is better as well.

**Comment by yaoyaoding** [comment]
Thanks @Aalanli !

**Comment by yaoyaoding** [comment]
Thanks @Aalanli !

**Pull request merged**